tests/components/device_automation/test_init.py

```diff
@@ -1128,12 +1128,14 @@ async def test_automation_with_integration_without_device_condition(
     )
 
 
+@pytest.mark.parametrize("trigger_key", ["platform", "trigger"])
```

Comment:
I don't think this parametrization should happen here. This should instead happen in `tests/components/automation/test_init.py` to make sure both the old and new keys work